lib/app-common.ts

```diff
@@ -115,21 +117,36 @@ export interface GlobalTunnelNgConfig {
  * default exclusion patterns are added for all private IPv4 address ranges.
  */
 async function setupGlobalHttpProxy(settings: CliSettings) {
-	const semver = await import('semver');
-	if (semver.lt(process.version, '10.16.0')) {
-		setupGlobalTunnelNgProxy(settings);
-	} else {
-		// use global-agent instead of global-tunnel-ng
-		await setupGlobalAgentProxy(settings);
+	// `global-tunnel-ng` accepts lowercase variables with higher precedence
+	// than uppercase variables, but `global-agent` does not accept lowercase.
+	// Set uppercase versions for backwards compatibility.
+	const { env } = process;
+	if (env.http_proxy) {
+		env.HTTP_PROXY = env.http_proxy;
+	}
+	if (env.https_proxy) {
+		env.HTTPS_PROXY = env.https_proxy;
+	}
+	delete env.http_proxy;
+	delete env.https_proxy;
+
+	const proxy = settings.getCatch<ProxyConfig>('proxy');
+	if (proxy || env.HTTPS_PROXY != null || env.HTTP_PROXY != null) {
```

Comment:
If I recall correctly, in some platform (Windows?) or shell (PowerShell?) it is possible to set an environment variable to the empty string (i.e. setting it in the shell, in a way that the CLI process sees it). However there's no need to load the proxy modules if an environment variable is set to the empty string, so I'd replace the `!= null` test with:
```
if (proxy || env.HTTPS_PROXY || env.HTTP_PROXY)
```
This would also be more consistent with the other tests a few lines above:
```
if (env.http_proxy) ...
if (env.https_proxy) ...
```

Reply:
No idea if you can do it in cmd/powershell but you can definitely do it in bash

Reply:
Changed